packages/custom_lint/lib/src/workspace.dart

```diff
@@ -429,8 +429,14 @@ Iterable<String> _findRoots(String path) sync* {
       return false;
     }
     // Check if the project has a package_config.json file.
-    return File(join(file.parent.path, '.dart_tool/package_config.json'))
-        .existsSync();
+    final isChildFromCache =
+        file.path.contains('.dart_tool') || file.path.contains('.symlinks');
```

Comment:
fwiw it's not safe to use `.contains`.
This would make against things like `path/foo.dart_tool` or `.dart_toolwhatever`

We should be using file.uri.segments.any